### DISCUSSION TITLE
Zigbee device short address is not constant thus my rules do not work.

### DISCUSSION THREAD

**Discussion by nmaou** [arendst/Tasmota#15035] [Q&A] [answered]
Hello,
I am using Tasmota on SonOFF zigbee bridge.
After pairing the zigbee devices I noticed that their short address changes randomly. The result of that is that the rules I use to publish MQTT messages do not work since the short address is changed so the rules are not triggered.
Is there a way to tell Tasmota to statically allocate the zigbee addresses;
The following are the rules I use to publish to a MQTT topic when a door security magnetic reed switch is activated.

  Rule2 
 on ZbReceived#0xdf9c#Contact=0 do publish NMAuto/02_IPNO/Open 0 endon
 on ZbReceived#0xdf9c#Contact=1 do publish NMAuto/02_IPNO/Open 1 endon

Any suggestions are welcomed,
Napoleon Maou

**Comment by nmaou** [comment]
I forgot to mention that the reed switches are Sonoff SNZB-04 ZigBee Wireless Door/Window Sensor.

**Comment by rebel7580** [comment, marked as answer]
Assign and use "FriendlyNames" (zbname) instead. These won't change; short addresses can, as you found out.

**Reply by nmaou** [reply]
Thanks for the reply
I tried using friendly names. It did not work. These are the rules I used. I show in the console that the switch opening was detected but the rule was not triggered..

Rule2 
 on ZbReceived#reed1#Contact=0 do publish NMAuto/02_IPNO/Open 0 endon
 on ZbReceived#reed1#Contact=1 do publish NMAuto/02_IPNO/Open 1 endon

**Reply by nmaou** [reply]
And this is the result of zbstatus. The reed switch I test is the last device. There is no difference in the name I use in the rules.

19:50:21.104 MQT: stat/tasmota_E2A592/RESULT = {"ZbStatus1":[{"Device":"0xF2EE","Name":" DOOR_SALONI"},{"Device":"0xC00F","Name":"SWITCH_01"},{"Device":"0x9E47","Name":"DOOR_KOUZINA"},{"Device":"0x4F11","Name":"DOOR_LIGHT_01"},{"Device":"0xF937","Name":"DOOR_MIKRO"},{"Device":"0xD97A","Name":"MOTION_EXTERN_SALONI"},{"Device":"0x5D21","Name":"SWITCH_02"},{"Device":"0x359D","Name":" SWITCH_03"},{"Device":"0xDF9C","Name":"reed1"}]}

**Reply by rebel7580** [reply]
Hmm. I don't have my system configured the same as yours wrt SetOptions, so I can't try this myself.  I'm not an expert on Rules, but that looks OK to me.  Maybe someone more experienced knows what's wrong.
Can you share the cosole output showing the sequence to see what the ZbReceived line is, etc.?

**Reply by nmaou** [reply]
This is the output when I open and then close the reed switch with the following rules
Rule2 
 on zbreceived#reed1#Contact=0 do publish NMAuto/02_IPNO/Open 0 endon
 on zbreceived#reed1#Contact=1 do publish NMAuto/02_IPNO/Open 1 endon


Output
22:28:41.378 RSL: SENSOR = {"ZbReceived":{"0xDF9C":{"Device":"0xDF9C","Name":"reed1","0500<00":"010000010000","ZoneStatusChange":1,"ZoneStatusChangeZone":1,"Contact":1,"Endpoint":1,"LinkQuality":102}}}
22:28:48.402 RSL: SENSOR = {"ZbReceived":{"0xDF9C":{"Device":"0xDF9C","Name":"reed1","0500<00":"000000010000","ZoneStatusChange":0,"ZoneStatusChangeZone":1,"Contact":0,"Endpoint":1,"LinkQuality":102}}}

This is the output when I use the rules with the shortaddr 
Rule2 
 on ZbReceived#0xdf9c#Contact=0 do publish NMAuto/02_IPNO/Open 0 endon
 on ZbReceived#0xdf9c#Contact=1 do publish NMAuto/02_IPNO/Open 1 endon

Output
22:30:28.242 MQT: stat/tasmota_E2A592/RESULT = {"Rule2":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":140,"Free":371,"Rules":"on ZbReceived#0xdf9c#Contact=0 do publish NMAuto/02_IPNO/Open 0 endon  on ZbReceived#0xdf9c#Contact=1 do publish NMAuto/02_IPNO/Open 1 endon"}}
22:30:32.605 MQT: tele/tasmota_E2A592/STATE = {"Time":"2022-03-05T22:30:32","Uptime":"17T07:09:22","UptimeSec":1494562,"Vcc":3.543,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":20,"Wifi":{"AP":1,"SSId":"NMAuto","BSSId":"74:EA:3A:DA:8D:04","Channel":4,"Mode":"11n","RSSI":100,"Signal":-26,"LinkCount":15,"Downtime":"0T00:06:26"}}
22:30:36.933 MQT: tele/tasmota_E2A592/SENSOR = {"ZbReceived":{"0xDF9C":{"Device":"0xDF9C","Name":"reed1","0500<00":"010000010000","ZoneStatusChange":1,"ZoneStatusChangeZone":1,"Contact":1,"Endpoint":1,"LinkQuality":107}}}
22:30:36.974 RUL: ZBRECEIVED#0XDF9C#CONTACT=1 performs "publish NMAuto/02_IPNO/Open 1"
22:30:36.978 MQT: NMAuto/02_IPNO/Open = 1
22:30:40.728 MQT: tele/tasmota_E2A592/SENSOR = {"ZbReceived":{"0xDF9C":{"Device":"0xDF9C","Name":"reed1","0500<00":"000000010000","ZoneStatusChange":0,"ZoneStatusChangeZone":1,"Contact":0,"Endpoint":1,"LinkQuality":102}}}
22:30:40.765 RUL: ZBRECEIVED#0XDF9C#CONTACT=0 performs "publish NMAuto/02_IPNO/Open 0"
22:30:40.769 MQT: NMAuto/02_IPNO/Open = 0

**Reply by nmaou** [reply]
And this is the output of the rule2 command that lists the rule
22:45:18.422 CMD: rule2
22:45:18.427 MQT: stat/tasmota_E2A592/RESULT = {"Rule2":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":138,"Free":373,"Rules":"on zbreceived#reed1#Contact=0 do publish NMAuto/02_IPNO/Open 0 endon  on zbreceived#reed1#Contact=1 do publish NMAuto/02_IPNO/Open 1 endon"}}

**Reply by rebel7580** [reply]
Your SENSOR report is still using  " {"ZbReceived":`{"0xDF9C"`" short address. Try SetOption83=1 to replace that with the Friendly name. 
That should give you {"ZbReceived":`{"reed1"`".  Then your Rule should work.

**Reply by nmaou** [reply]
It worked like a charm. Thanks for the help.

**Comment by rebel7580** [comment]
please close. thx.

**Reply by nmaou** [reply]
I do not know how to close the discussion. How do I do it?

**Reply by rebel7580** [reply]
On the post with the "answer", there should be a "button" to click called "Mark as Answer" -- I think!
